docs/source/reference/server-api.md

```diff
@@ -0,0 +1,367 @@
+# Starting servers with the JupyterHub API
+
+JupyterHub's [REST API][] allows launching.
+This allows you to build services launching Jupyter-based services for users
+without relying on the JupyterHub API at all.
+[BinderHub][] is an example of such an application.
+
+[binderhub]: https://binderhub.readthedocs.io
+[rest api]: ../reference/rest.md
+
+This document provides an example of working with the JupyterHub API to
+manager servers for users.
+In particular, we will cover how to:
+
+1. [check status of servers](checking)
+2. [start servers](starting)
+3. [wait for servers to be ready](waiting)
+4. [communicate with servers](communicating)
+5. [stop servers](stopping)
+
+(checking)=
+
+## Checking server status
+
+Requesting information about a user includes a `servers` field,
+which is a dictionary.
+
+```
+GET /hub/api/users/:username
+```
+
+**Required scope: `read:servers`**
+
+```python
+{
+  "admin": false,
+  "groups": [],
+  "pending": null,
+  "server": null,
+  "name": "test-1",
+  "kind": "user",
+  "last_activity": "2021-08-03T18:12:46.026411Z",
+  "created": "2021-08-03T18:09:59.767600Z",
+  "roles": ["user"],
+  "servers": {}
+}
+```
+
+If the `servers` dict is empty, the user has no running servers.
+The keys of the `servers` dict are server names as strings.
+Many JupyterHub deployments only use the 'default' server,
+which has the empty string `''` for a name.
+In this case, the servers dict will always have either zero or one elements.
+
+This is the servers dict when the user's default server is fully running and ready:
+
+```python
```

Comment:
```suggestion
```json
```

Reply:
I actually did Python on purpose because the lexer is more lax (e.g. trailing commas fail JSON parsing, and can't insert comments in JSON)